Source/spinal-harmony.cs

```diff
@@ -1,314 +1,327 @@
-    public class PatchEngineFuel
-    {
-        public static void Postfix(ref CompProperties_EngineTrail __result, CompEngineTrail __instance)
-        {
-            // Log.Message("SOS2ExpSE Postfix enginetrail props called");
-            if (__instance.parent.def.defName == "Ship_Engine_Spinal") // This feels filthy I bet there is a better way TODO
+                return true; //building is already in cache, presumably re-running on other cells
+            }
+            if (argBuilding.TryGetComp<CompSpinalEngineMount>() == null) 
             {
-                float tempAmpBonus = SOS2ExpSpinalEngines.SpinalRecalc(__instance);
-                if (tempAmpBonus != 0)
-                {
-                    Log.Message($"Spinal Engine fuel (OG {__result.fuelUse}) postfixed value (Bonus {tempAmpBonus}) is: {__result.fuelUse * (1 + tempAmpBonus)}");
-                    // __result.fuelUse = (int)(__result.fuelUse * (1 + tempAmpBonus));
-                }
-                else // Incomplete spinal engine, no fuel use
-                {
-                     // __result.fuelUse = 0;
-                }
-            } // postfix so no "else", already correct value
-        }
-    }
-    [HarmonyPatch(typeof(SaveOurShip2.DoSpawn), nameof(SaveOurShip2.DoSpawn.OnSpawn), MethodType.Normal)] // postfixing a postfix feels wrong 
-    public class PatchSpinalSpawn
-    {
-       public static void Postfix(object[] __args)
-        {
-            if ((bool)__args[2])
-                return;
-            if (__args[0] == null || __args[1] == null || __args[2] == null)
+                //Log.Message("SOS2ExpSpinalEngines SpinalComp null");
+                return true; //Not spinal-related, bail ASAP
+            }
+            Building foundEngine = TheCafFiend.SOS2SpinalEngines.EngineFromSpinal(argBuilding);
+            if (foundEngine == null)
             {
-                Log.Error("Did something change in SOS2OnSpawn? Invalid for patching from SOS2ExpSpinalEngines");
-                return;
+                //Log.Message("SOS2 SpinalEngines EngineFromSpinal null");
+                return true; //amp/cap but no engine
             }
-            // Can't say I care for all this but OnSpawn is static so I gotta steal the args and rebuild half of it apparently
-            Map argMap = (Map)__args[1];
-            Building argBuilding = (Building)__args[0];
-            // Log.Message("SOS2ExpSE Postfix Postfix (lol) spawn called");
-            var mapComp = argMap.GetComponent<ShipHeatMapComp>();
-            if (mapComp.CacheOff || mapComp.ShipsOnMapNew.NullOrEmpty())
+            CompSpinalEngineTrail foundEngineComp = foundEngine.TryGetComp<CompSpinalEngineTrail>();
+            if (foundEngineComp == null)
             {
-                return;
+                Log.Error($"SOS2Spinal engines foundEngineComp is null! foundengine {foundEngine.def.defName}");
+                return true; // I guess? true? 
```

Comment:
True here is related to harmony (sorcery reflection helper library)  magic due to the fact it's a prefix: This runs *before* the function it is modifying, and if it ever returns false it dumps any other prefixes (This is potentially undefined behaviour because other prefixes, depending on the users mod order IIRC, may be before or! after yours) and also bails on the entire function. I was debating if I should return true here or not because something went kinda wrong but I didn't want to break the shipcache because something went wrong, basically